### ISSUE TITLE
Adding ICC profile to PDF

### ISSUE THREAD

**Issue by joseavegaa** [pikepdf/pikepdf#477]
I know this isn't the place to ask for help, but I really can't find any answers. 

I have a PDF file and I am able to open it using pikepdf, and edit the metadata. I am able to correct everything and get a validated PDF/1A with Verapdf. 

I am trying to generate a PDF/4 (2.0 standard), and that requires to add an ICC profile. That's the only thing that I am missing according to Verapdf. I have an ICC profile ready (sRGB) but I do not see a way to add it to the existing PDF using pikepdf. 

Basically, I need a way to add the ICC profile to the PDF. Can you provide some guidance on the issue? Thanks.

**Comment by jbarlow83** [comment]
Have a look at ocrmypdf's pdfa.py file - it generates a little bit of Postscript and uses Ghostscript to attach an ICC to the PDF/A. (The same could be done with pikepdf directly, but Ghostscript needs this as an input to convert an arbitrary PDF to PDF/A.) This should be enough to illustrate where it should be attached.

PDF/A needs an output intent ICC profile which is different from individual ICC profiles that might be attached to images or an ExtGState object.
